org-code-javabuilder/lib/src/main/java/org/code/javabuilder/JavaBuilder.java

```diff
@@ -0,0 +1,71 @@
+package org.code.javabuilder;
+
+import com.amazonaws.client.builder.AwsClientBuilder;
+import com.amazonaws.services.apigatewaymanagementapi.AmazonApiGatewayManagementApi;
+import com.amazonaws.services.apigatewaymanagementapi.AmazonApiGatewayManagementApiClientBuilder;
+import com.amazonaws.services.sqs.AmazonSQS;
+import com.amazonaws.services.sqs.AmazonSQSClientBuilder;
+import java.io.IOException;
+
+public class JavaBuilder {
+  private final OutputHandler outputHandler;
+  private final InputPoller inputPoller;
+  private final OutputPoller outputPoller;
+  private final JavaRunner javaRunner;
+  private final OutputSemaphore outputSemaphore;
+
+  public JavaBuilder(String connectionId, String apiEndpoint, String queueUrl) {
+    // Create output handler
+    AmazonApiGatewayManagementApi api =
+        AmazonApiGatewayManagementApiClientBuilder.standard()
+            .withEndpointConfiguration(
+                new AwsClientBuilder.EndpointConfiguration(apiEndpoint, "us-east-1"))
+            .build();
+    this.outputHandler = new OutputHandler(connectionId, api);
+    this.outputSemaphore = new OutputSemaphore();
+
+    // Overwrite system I/O
+    RuntimeIO runtimeIO;
+    try {
+      runtimeIO = new RuntimeIO(this.outputSemaphore);
+    } catch (IOException e) {
+      this.outputHandler.sendMessage("There was an error running your code. Try again.");
+      throw new RuntimeException("Error setting up console IO", e);
+    }
+
+    // Create code runner
+    this.javaRunner = new JavaRunner(this.outputSemaphore);
+
+    // Create input poller
+    AmazonSQS sqsClient = AmazonSQSClientBuilder.defaultClient();
+    this.inputPoller =
+        new InputPoller(sqsClient, queueUrl, runtimeIO, this.javaRunner, this.outputHandler);
+
+    // Create output poller
+    this.outputPoller =
+        new OutputPoller(this.javaRunner, this.outputHandler, runtimeIO, this.outputSemaphore);
+  }
+
+  public void runUserCode() {
+    this.javaRunner.start();
+    this.inputPoller.start();
+    this.outputPoller.start();
+    while (javaRunner.isAlive()) {
```

Comment:
I'm unsure how to get around this instance of busy-waiting. We can't let LambdaRequestHandler.handleRequest return until the student's code is finished running. Tips? Ideas?